cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -1063,6 +1063,10 @@ class CategoricalBlock : public PandasBlock {
     using T = typename TRAITS::T;
     constexpr int npy_type = TRAITS::npy_type;
 
+    if (data->num_chunks() <= 0) {
```

Comment:
nit: num_chunks() cannot be negative (even though it's a signed integer), so checking `== 0` is better